x/gov/proposals.go

```diff
@@ -9,142 +9,98 @@ import (
 	sdk "github.com/cosmos/cosmos-sdk/types"
 )
 
-// Proposal interface
-type Proposal interface {
-	GetProposalID() uint64
-	SetProposalID(uint64)
+// Proposal is a struct used by gov module internally
+// embedds ProposalContent with additional fields to record the status of the proposal process
+type Proposal struct {
+	ProposalContent `json:"proposal_content"` // Proposal interface
 
-	GetTitle() string
-	SetTitle(string)
-
-	GetDescription() string
-	SetDescription(string)
-
-	GetProposalType() ProposalKind
-	SetProposalType(ProposalKind)
-
-	GetStatus() ProposalStatus
-	SetStatus(ProposalStatus)
-
-	GetFinalTallyResult() TallyResult
-	SetFinalTallyResult(TallyResult)
+	ProposalID uint64 `json:"proposal_id"` //  ID of the proposal
 
-	GetSubmitTime() time.Time
-	SetSubmitTime(time.Time)
-
-	GetDepositEndTime() time.Time
-	SetDepositEndTime(time.Time)
+	Status           ProposalStatus `json:"proposal_status"`    //  Status of the Proposal {Pending, Active, Passed, Rejected}
+	FinalTallyResult TallyResult    `json:"final_tally_result"` //  Result of Tallys
 
-	GetTotalDeposit() sdk.Coins
-	SetTotalDeposit(sdk.Coins)
+	SubmitTime     time.Time `json:"submit_time"`      //  Time of the block where TxGovSubmitProposal was included
+	DepositEndTime time.Time `json:"deposit_end_time"` // Time that the Proposal would expire if deposit amount isn't met
+	TotalDeposit   sdk.Coins `json:"total_deposit"`    //  Current deposit on this proposal. Initial value is set at InitialDeposit
 
-	GetVotingStartTime() time.Time
-	SetVotingStartTime(time.Time)
+	VotingStartTime time.Time `json:"voting_start_time"` //  Time of the block where MinDeposit was reached. -1 if MinDeposit is not reached
+	VotingEndTime   time.Time `json:"voting_end_time"`   // Time that the VotingPeriod for this proposal will end and votes will be tallied
+}
 
-	GetVotingEndTime() time.Time
-	SetVotingEndTime(time.Time)
+// nolint
+func (p Proposal) String() string {
+	return fmt.Sprintf(`Proposal %d:
+		  Title:              %s
+		  Type:               %s
+		  Status:             %s
+		  Submit Time:        %s
+		  Deposit End Time:   %s
+		  Total Deposit:      %s
+		  Voting Start Time:  %s
+		  Voting End Time:    %s`, p.ProposalID, p.GetTitle(), p.ProposalType(),
+		p.Status, p.SubmitTime, p.DepositEndTime,
+		p.TotalDeposit, p.VotingStartTime, p.VotingEndTime)
+}
 
-	String() string
+// ProposalContent is an interface that has title, description, and proposaltype for the Proposal
```

Comment:
Let's update this comment - AFAICT the struct which fulfills `ProposalContent` may also contain more detailed specifications as to how the proposal is to be executed (ex. the parameter set change amount) - we should explain this in this comment here